instrumentation/opentelemetry_ecto/lib/opentelemetry_ecto.ex

```diff
@@ -125,14 +125,43 @@ defmodule OpentelemetryEcto do
 
     OpenTelemetry.Span.end_span(s)
 
-    if parent_context != :undefined do
-      OpenTelemetry.Ctx.detach(parent_context)
-    end
+    if ctx?(parent_context), do: OpenTelemetry.Ctx.detach(parent_context)
   end
 
   defp format_error(%{__exception__: true} = exception) do
     Exception.message(exception)
   end
 
   defp format_error(_), do: ""
+
+  defp ctx?(:undefined), do: false
+  defp ctx?(_), do: true
+
+  defp maybe_add(map, _, nil), do: map
+  defp maybe_add(map, key, value) when is_map(map), do: Map.put(map, key, value)
+
+  defp maybe_append_source(name, nil), do: name
+  defp maybe_append_source(name, source), do: "#{name}:#{source}"
+
+  defp maybe_add_peer_addr(map, peer_name) do
+    if ip?(peer_name), do: maybe_add(map, :"net.sock.peer.addr", peer_name), else: map
+  end
+
+  defp db_system(Ecto.Adapters.Postgres), do: "postgresql"
+  defp db_system(Ecto.Adapters.MyXQL), do: "mysql"
+  defp db_system(Ecto.Adapters.Tds), do: "mssql"
+  defp db_system(other) when is_atom(other), do: other |> Atom.to_string() |> String.downcase()
```

Comment:
This is not very performant but it would trim the module to the last bit.

```suggestion
  defp db_system(other) when is_atom(other), do: other |> Module.split() |> List.last() |> String.downcase()
```

Ideally, PRs will just be created for any other databases since the list is well-defined in the spec and is only supposed to fall back to a custom value when it isn't in the list.